rasa/core/slots.py

```diff
@@ -198,6 +199,10 @@ def __init__(
         super().__init__(name, initial_value, value_reset_delay, auto_fill)
         self.values = [str(v).lower() for v in values] if values else []
 
+    def add_default_value(self) -> None:
+        if DEFAULT_CATEGORICAL_SLOT_VALUE not in self.values:
```

Comment:
That's a little bit out of this task, by did we consider making self.values a Set instead of a List? Then this search operation will be simply faster. Of course, only if values there ARE unique-only.
I'm not sure though how many values are usually there, maybe it's a nano optimisation :)

Reply:
I think they should be unique. A user could define a categorical slot with thousands of values, but most likely there would be a few dozen at most.